test/unit/host_test.rb

```diff
@@ -1367,6 +1367,10 @@ def teardown
       assert_equal hosts.count, 1
       assert_equal ["num001.example.com"], hosts.map { |h| h.name }.sort
 
+      hosts = Host::Managed.search_for("facts.memory_mb ~ 64498")
```

Comment:
This test is passing on the current develop branch, without the change above in `scoped_search_extensions`. Could you add some test that tests the functionality added here?